internal/dao/pod.go

```diff
@@ -455,3 +457,40 @@ func in(ll []string, s string) bool {
 	}
 	return false
 }
+
+func (p *Pod) GetPodSpec(path string) (*v1.PodSpec, error) {
+	pod, err := p.GetInstance(path)
+	if err != nil {
+		return nil, err
+	}
+	podSpec := pod.Spec
+	return &podSpec, nil
+}
+
+func (p *Pod) SetImages(ctx context.Context, path string, spec v1.PodSpec) error {
```

Comment:
I think we have to be careful here. If the pod is unmanaged ie plain ol' pod I think that's ok. However if the pod is managed via dp, sts, ... setting the image on a single pod could be dangerous and could yield hard to trouble shoot situations. So I think that option should only be available on unmanaged pods. Also if k9s is started in readonly mode we need to disable the set image options on all resources.

Reply:
This is the reason why I am not a big fan of one offs cluster changes as image updates are better managed via manifests as this could yield hard to debug issues on live clusters with the way to see who changed what.

Reply:
Pod is managed by a controller if it has `metadata.ownerReferences[*].controller == true`

Reply:
Yes that is correct Antoine! Actually if that slice is not empty we could issue a message saying something like: "Unable to set image. This pod is managed. Please set the image on the controller". If you want to go to town, you could pull the  first controller type/name and display it.  Would this make sense?

Reply:
I return an error with controller name, after a user validate set image form. 

<img width="1440" alt="Capture d’écran 2020-09-28 à 13 59 47" src="https://user-images.githubusercontent.com/592978/94429744-16d7e500-0193-11eb-827e-ba1f1334d6eb.png">